### ISSUE TITLE
Invalid caret position for missing given argument error messages

### ISSUE THREAD

**Issue by som-snytt** [scala/scala3#15630]
## Compiler version

Welcome to Scala 3.2.1-RC1-bin-SNAPSHOT-git-baf21dd (18.0.1.1, Java OpenJDK 64-Bit Server VM).

## Minimized code

```Scala
scala> def f = summon[Int]
-- Error: --------------------------------------------------------------------------------------------------------------
1 |def f = summon[Int]
  |                   ^
  |                   No given instance of type Int was found for parameter x of method summon in object Predef
1 error found
```

## Output

The caret doesn't seem to be aligned.

## Expectation

The caret should be aligned with the guilty text.

**Comment by griggt** [comment]
It seems that the caret is pointing to the missing given argument.  This may be by design, IDK.

**Comment by som-snytt** [comment]
@griggt Clever! So the caret is really a pen, and the compiler can't think of anything to write.

**Comment by som-snytt** [comment]
Yes, at the paren https://github.com/lampepfl/dotty/pull/15616/files#diff-e19e7d425dad1356477e831edebb088a7560425354d2061fe4f4feeb0e3f68beR3

and no paren https://github.com/lampepfl/dotty/pull/15616/files#diff-e19e7d425dad1356477e831edebb088a7560425354d2061fe4f4feeb0e3f68beR34

This may be a feature request. 😄

**Comment by som-snytt** [comment]
Only I found it confusing.
